### PR TITLE
Improve errors from frame.waitFor* methods

### LINKED ISSUE #394: Improve errors from frame.waitFor* methods
https://github.com/kblok/puppeteer/pull/3/commits/dde45faaebd4232ef953622a8e1e07e544cf64a8

### PR DESCRIPTION
**Pull request by kblok**
closes #394

### CODE REVIEW

File: lib/PuppeteerSharp.Tests/FrameTests/WaitForSelectorTests.cs [hunk 1]
```diff
@@ -175,11 +175,11 @@ public WaitForSelectorTests(ITestOutputHelper output) : base(output)
         [Fact]
         public async Task ShouldRespectTimeout()
         {
-            var exception = await Assert.ThrowsAnyAsync<PuppeteerException>(async ()
+            var exception = await Assert.ThrowsAnyAsync<WaitTaskTimeoutException>(async ()
```
Comment (Meir017) [inline lib/PuppeteerSharp.Tests/FrameTests/WaitForSelectorTests.cs]: use `ThrowsAsync` instead

**Comment by kblok** [comment]
Done @Meir017

File: lib/PuppeteerSharp/WaitTaskTimeoutException.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace PuppeteerSharp
+{
+    [Serializable]
+    internal class WaitTaskTimeoutException : Exception
```
Comment (Meir017) [inline lib/PuppeteerSharp/WaitTaskTimeoutException.cs]: base class should be `PuppeteerException`

Comment (kblok) [reply lib/PuppeteerSharp/WaitTaskTimeoutException.cs]: @Meir017 genius. Done.

**Comment by kblok** [comment]
@Meir017 could you take a look at the latest commit?

File: lib/PuppeteerSharp/Frame.cs [hunk 3]
```diff
@@ -262,15 +262,27 @@ public Task<ElementHandle> WaitForXPathAsync(string xpath, WaitForSelectorOption
...
+
+        /// <summary>
+        /// Waits for an expression to be evaluated to a truthy value
+        /// </summary>
+        /// <param name="script">Expression to be evaluated in browser context</param>
+        /// <param name="options">Optional waiting parameters</param>
+        /// <param name="args">Arguments to pass to <c>script</c></param>
+        /// <returns>A task that resolves when the <c>script</c> returns a truthy value</returns>
+        /// <seealso cref="Page.WaitForExpressionAsync(string, WaitForFunctionOptions, object[])"/>
+        public Task<JSHandle> WaitForExpressionAsync(string script, WaitForFunctionOptions options, params object[] args)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Frame.cs]: expressions don't have arguments, I think you should remove the `args` argument

**Comment by kblok** [comment]
Done @Meir017

**Pull request merged**